ent/encryption/encryption.cc

```diff
@@ -823,6 +823,21 @@ class encryption_file_io_extension : public sstables::file_io_extension {
         });
         co_return sink;
```

Comment:
A lot more seems to be happening here, than what the commitlog suggests. Please split of the propagation of `stream_creator_fn` into a separate commit with its own explanation.

Reply:
Done